### PR TITLE
Subpackages

### PR DESCRIPTION
**Pull request by sjsrey**
Before this pr our import time is:
```
(analytical) src/pysal - [nolazy●] » time python -c "import pysal"
python -c "import pysal"  2.92s user 0.45s system 96% cpu 3.509 total
```
After this pr it becomes:
```
(analytical) src/pysal - [subpackages●] » time python -c "import pysal"
python -c "import pysal"  0.04s user 0.01s system 74% cpu 0.064 total
```

It also mimics the approach used in [scipy](https://github.com/scipy/scipy/blob/master/scipy/__init__.py) where subpackages require an explicit import.

### CODE REVIEW

File: ci/36.yml [hunk 1]
```diff
@@ -3,21 +3,20 @@ channels:
...
+  # testing
   - matplotlib
-  - descartes
-  - palettable
-  - statsmodels
+  - pytest
+  - pytest-cov
+  - coverage
+  - codecov
+  - coveralls
```
Comment (jGaboardi) [inline ci/36.yml:16]: We can remove `coveralls` and `nose`.

File: ci/37.yml [hunk 2]
```diff
@@ -3,20 +3,22 @@ channels:
...
+  - deprecated
+  # testing
   - matplotlib
-  - descartes
-  - palettable
+  - pytest
+  - pytest-cov
+  - coverage
+  - codecov
+  - coveralls
```
Comment (jGaboardi) [inline ci/37.yml:15]: We can remove `coveralls` and `nose`.

File: ci/38.yml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  - six
+  - pyyaml
+  - deprecated
+  # testing
+  - codecov
+  - matplotlib
+  - pytest
+  - pytest-cov
+  - coverage
+  - coveralls
```
Comment (jGaboardi) [inline ci/38.yml:15]: We can remove `coveralls` and `nose`.

File: pysal/base.py [hunk 4]
```diff
@@ -20,6 +20,7 @@
                'pointpats': 'explore',
                'inequality': 'explore',
                'spaghetti': 'explore',
+               'access': 'access',
```
Comment (knaaptime) [inline pysal/base.py:23]: should this be `'access: 'explore'`?

File: README.md [hunk 5]
```diff
@@ -61,6 +61,10 @@ In contrast to `explore`, the `model` layer focuses on confirmatory analysis. In
 
 -   [tobler](http://pysal.org/tobler/) : `tobler` provides functionality for for areal interpolation and dasymetric mapping. Its name is an homage to the legendary geographer Waldo Tobler a pioneer of dozens of spatial analytical methods. `tobler` includes functionality for interpolating data using area-weighted approaches, regression model-based approaches that leverage remotely-sensed raster data as auxiliary information, and hybrid approaches.
 
+-   [access](http://github.com/pysal/access) : `access` aims to make it easy for analysis to calculate measures of spatial accessibility. this work has traditionally had two challenges: [1] to calculate accurate travel time matrices at scale and [2] to derive measures of access using the travel times and supply and demand locations. `access` implements classic spatial access models, allowing easy comparison of methodologies and assumptions.
```
Comment (jGaboardi) [inline README.md:64]: The sentences have been uncapitalized again.

**Comment by jGaboardi** [comment]
We may want to go ahead and disable `appveyor`.

**Comment by sjsrey** [comment]
> We may want to go ahead and disable `appveyor`.

I did remove appveyor. It passes on [my remote](https://github.com/sjsrey/pysal/pull/58) but not here. I'm wondering why this is?

**Comment by jGaboardi** [comment]
> > We may want to go ahead and disable `appveyor`.
> 
> I did remove appveyor. It passes on [my remote](https://github.com/sjsrey/pysal/pull/58) but not here. I'm wondering why this is?

It passes on your remote because it is part of your `master` and GHA is triggered for your repo. Since it has not been merged with the `pysal/master`, `appveyor` is still triggered and GHA is not yet being triggered.

**Pull request merged**